Sources/SourceKit/sourcekitd/SwiftLanguageServer.swift

```diff
@@ -131,6 +148,27 @@ extension SwiftLanguageServer {
       guard let self = self else { return }
       let notification = SKResponse(notification, sourcekitd: self.sourcekitd)
 
+      if notification.value?[self.keys.notification] == self.values.notification_sema_enabled {
+        self.state = .connected
```

Comment:
The notification handler queue is not related to `queue` for this object, so to access `state` we need to synchronize. Similarly, `addStateChangeHandler` needs to synchronize the same way.

Reply:
As I understood it, `queue` is used to synchronise the requests and responses from `sourcekitd`. Setting the state does not fall into this category, so I did not synchronise it on `queue`. 

Do you want to extend the definition of `queue`? I don’t think that `state` really needs to be synchronised, but it wouldn’t hurt either.

Reply:
The comment on `queue` is misleading and should be updated.  We need to protect shared access to mutable state - e.g. we're doing this for `currentDiagnostics` already.

Reply:
OK, in that case, it makes total sense to guard the access to state by `queue`.